cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -0,0 +1,208 @@
+      log.error(e.getMessage());
+      if (((HttpClientErrorException.Conflict) e).getStatusCode().value() == 409) {
+        return ApiResponse.builder()
+            .result("Schema being registered is incompatible with an earlier schema")
+            .build();
+      }
+      return ApiResponse.builder().result("Failure in registering schema.").build();
+    }
+  }
+
+  public Map<Integer, Map<String, Object>> getSchema(
+      String environmentVal, KafkaSupportedProtocol protocol, String topicName) {
+    try {
+      log.info("Into getSchema request {} {} {}", topicName, environmentVal, protocol);
+      if (environmentVal == null) {
+        return null;
+      }
+
+      List<Integer> versionsList = getSchemaVersions(environmentVal, topicName, protocol);
+      String schemaCompatibility = getSchemaCompatibility(environmentVal, topicName, protocol);
+      Map<Integer, Map<String, Object>> allSchemaObjects = new TreeMap<>();
+
+      if (versionsList != null) {
+        for (Integer schemaVersion : versionsList) {
+          String suffixUrl =
+              environmentVal + "/subjects/" + topicName + "-value/versions/" + schemaVersion;
+          Pair<String, RestTemplate> reqDetails =
+              clusterApiUtils.getRequestDetails(
+                  suffixUrl, protocol, KafkaClustersType.SCHEMA_REGISTRY);
+
+          Map<String, String> params = new HashMap<>();
+
+          ResponseEntity<HashMap> responseNew =
+              reqDetails.getRight().getForEntity(reqDetails.getLeft(), HashMap.class, params);
+          Map<String, Object> schemaResponse = responseNew.getBody();
+          if (schemaResponse != null) {
+            schemaResponse.put("compatibility", schemaCompatibility);
+          }
+
+          log.info(Objects.requireNonNull(responseNew.getBody()).toString());
+          allSchemaObjects.put(schemaVersion, schemaResponse);
+        }
+      }
+
+      return allSchemaObjects;
+    } catch (Exception e) {
+      log.error("Error from getSchema : " + e.getMessage());
+      return new TreeMap<>();
+    }
+  }
+
+  private List<Integer> getSchemaVersions(
+      String environmentVal, String topicName, KafkaSupportedProtocol protocol) {
+    try {
+      log.info("Into getSchema versions {} {}", topicName, environmentVal);
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String suffixUrl = environmentVal + "/subjects/" + topicName + "-value/versions";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.SCHEMA_REGISTRY);
+
+      Map<String, String> params = new HashMap<>();
+
+      ResponseEntity<ArrayList> responseList =
+          reqDetails.getRight().getForEntity(reqDetails.getLeft(), ArrayList.class, params);
+      log.info("Schema versions " + responseList);
+      return responseList.getBody();
+    } catch (Exception e) {
+      log.error("Error in getting versions " + e.getMessage());
```

Comment:
```suggestion
      log.error("Error in getting versions ", e);
```
it's better to log the whole trace